lib/views/mobile/widgets/PreferenceEditor.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:streaming_shared_preferences/streaming_shared_preferences.dart';
+
+import '../../../constants/app_constants.dart';
+import '../../../constants/asset_path.dart';
+import '../../widgets/Heading.dart';
+import 'buttons/Button.dart';
+
+List<String> singleDefault(String defaultValue) {
+  return [defaultValue];
+}
+
+class PopupEditText extends StatefulWidget {
+  final Preference<String> preference;
+  final List<String> Function(String) expandDefaults;
+
+  final String title;
+  final double marginTop;
+
+  PopupEditText(this.preference,
+      {this.marginTop = 0.0,
+      this.title = "",
+      this.expandDefaults = singleDefault});
+
+  @override
+  State<StatefulWidget> createState() => _PopupEditTextState(this.preference,
+      expandDefaults: this.expandDefaults,
+      marginTop: this.marginTop,
+      title: this.title);
+}
+
+class _PopupEditTextState extends State<PopupEditText> {
+  final Preference<String> preference;
+  late String value;
+  late List<String> defaultValues;
+
+  final title;
+  final double marginTop;
+
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+
+  final TextEditingController _textEditingController = TextEditingController();
+
+  _PopupEditTextState(this.preference,
+      {required this.marginTop,
+      required this.title,
+      required List<String> Function(String) expandDefaults})
+      : this.value = preference.getValue(),
+        this.defaultValues = expandDefaults(preference.defaultValue);
+
+  Future<void> showInformationDialog(BuildContext context) async {
+    var theme = Theme.of(context);
+    final List<Widget> defaultButtons = defaultValues
+        .map((value) => ElevatedButton(
+            onPressed: () {
+              update(value);
+              _textEditingController.text = value;
+            },
+            child: Text(value,
+                style: theme.textTheme.bodyText2?.copyWith(fontSize: 11.0))))
+        .toList();
+
+    return await showDialog(
+        context: context,
+        builder: (context) {
+          bool isChecked = false;
+          return StatefulBuilder(builder: (context, setState) {
+            return AlertDialog(
+              backgroundColor: theme.scaffoldBackgroundColor,
+              content: Form(
+                  key: _formKey,
+                  child: Column(
+                    mainAxisSize: MainAxisSize.min,
+                    children: [
+                          Text("Value:"),
+                          TextFormField(
```

Comment:
We should add length validation up to 100 or 150 symbols (couldn't find what maximum values we support for example for AppId).